Code/cmake/Modules/RDKitUtils.cmake

```diff
@@ -103,17 +103,25 @@ macro(rdkit_python_extension)
   if(RDK_BUILD_PYTHON_WRAPPERS)
     PYTHON_ADD_MODULE(${RDKPY_NAME} ${RDKPY_SOURCES})
     set_target_properties(${RDKPY_NAME} PROPERTIES PREFIX "")
-if(WIN32)
-    set_target_properties(${RDKPY_NAME} PROPERTIES SUFFIX ".pyd"
-                          LIBRARY_OUTPUT_DIRECTORY
-                          ${RDK_PYTHON_OUTPUT_DIRECTORY}/${RDKPY_DEST})
-else(WIN32)
-    set_target_properties(${RDKPY_NAME} PROPERTIES
-                          LIBRARY_OUTPUT_DIRECTORY
-                          ${RDK_PYTHON_OUTPUT_DIRECTORY}/${RDKPY_DEST})
-endif(WIN32)
-    target_link_libraries(${RDKPY_NAME} ${RDKPY_LINK_LIBRARIES}
-                          ${PYTHON_LIBRARIES} ${Boost_LIBRARIES} )
+    if(WIN32)
+      set_target_properties(${RDKPY_NAME} PROPERTIES SUFFIX ".pyd"
+                           LIBRARY_OUTPUT_DIRECTORY
+                           ${RDK_PYTHON_OUTPUT_DIRECTORY}/${RDKPY_DEST})
+    else(WIN32)
+        set_target_properties(${RDKPY_NAME} PROPERTIES
+                              LIBRARY_OUTPUT_DIRECTORY
+                              ${RDK_PYTHON_OUTPUT_DIRECTORY}/${RDKPY_DEST})
+    endif(WIN32)
+    
+    if("${Py_ENABLE_SHARED}" STREQUAL "1")
+      target_link_libraries(${RDKPY_NAME} ${RDKPY_LINK_LIBRARIES}
+                            ${PYTHON_LIBRARIES} ${Boost_LIBRARIES} )
+    else()
+      target_link_libraries(${RDKPY_NAME} ${RDKPY_LINK_LIBRARIES}
+                            ${Boost_LIBRARIES} )
+      message("set_target_properties ${RDKPY_NAME} PROPERTIES LINK_FLAGS ${PYTHON_LDSHARED}")                            
+      set_target_properties(${RDKPY_NAME} PROPERTIES LINK_FLAGS ${PYTHON_LDSHARED})
```

Comment:
on windows `${PYTHON_LDSHARED}` is empty, so this generates an error:
```
CMake Error at Code/cmake/Modules/RDKitUtils.cmake:123 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
```
You need the same test that you used for RDBoost